jupyter_client/manager.py

```diff
@@ -377,6 +377,9 @@ def shutdown_kernel(self, now=False, restart=False):
             self.finish_shutdown()
 
         self.cleanup(connection_file=not restart)
+        # shutdown ZMQ context as we no longer using this kernel
+        if not restart:
+            self.context.destroy(linger=100)
```

Comment:
Since it's a resource cleanup thing, maybe it should be inside `cleanup()`? Maybe we should change it to `cleanup(restart=False)` and have `connection_file = not restart` inside?

Reply:
This seems like a good idea.  If moved, I think we'd need to preserve the default value (e.g., `cleanup(shutdown=True)`). Even though it's associated with a different name, the semantics would be preserved and subclasses wouldn't require updates.

Reply:
+1

Reply:
Sure.